Add step-by-step guide for attaching PDFs to Chocolate library #14646

TimeTriggerx opened the pull request:
Closes JabRef/jabref#14646

I added a comprehensive, step-by-step guide (**HOW-TO-ATTACH-PDFS.md**) along with supporting screenshots inside the `chocolate` library folder. These changes provide clear, visual instructions for newcomers on how to attach research PDFs to JabRef entries, which is essential for properly populating the library and for using advanced features such as the AI **“Chat with group”** functionality.

---

## Steps to test

1. **Navigate to the directory**  
   Go to the `chocolate/` folder in this repository.

2. **Review the guide**  
   Open `README.md` and review the instructions for both:
   - Drag-and-Drop attachment  
   - Manual attachment using the right-click context menu and file selection

3. **Verify visuals**  
   Verify that the screenshots located in `chocolate/screenshots/` render correctly and accurately represent the JabRef interface.

4. **Check AI context**  
   Review the **Pro Tip** section and confirm it clearly explains the benefit of linking multiple PDFs for improved AI-based analysis.

---

## Screenshots of the process

### Drag and Drop Method
<img width="2944" height="1440" alt="Drag and Drop Method" src="https://github.com/user-attachments/assets/b31ee6bb-ebe6-4483-8fde-24e42c360793" />

### Manual Attachment (Context Menu & File Selection)
<img width="2848" height="1504" alt="Manual Attachment" src="https://github.com/user-attachments/assets/bfc8d4da-3a84-471a-b975-bb3480aee2c7" />

---

## Mandatory checks

- [x] I own the copyright of the code submitted and I license it under the MIT license  
- [x] I manually tested my changes in running JabRef (always required)  
- [/] I added JUnit tests for changes (if applicable)  
- [x] I added screenshots in the PR description (if change is visible to the user)  
- [/] I described the change in `CHANGELOG.md` in a way that is understandable for the average user (not strictly applicable for this documentation-focused change)  
- [x] I checked the user documentation: Is the information available and up to date?

TimeTriggerx committed:
Add documentation and screenshots for attaching PDFs #14646

koppor reviewed (changes requested):
1. Convert to markdown :)
2. Place it in `README.md`
3. Place images in `assets` folder in root (you need to create it)

Reason: When JabRef executes "find unlinked files" in choco sub folder, it will also see the images, which could be a WTF for new users.

TimeTriggerx committed:
Refactor: Move guide to root README and images to assets folder as requested

TimeTriggerx commented:
Hi @koppor, I've updated the PR based on your feedback. I have:

1. Moved the images to a root /assets folder to avoid interference with JabRef's 'find unlinked files' feature.

2. Merged the step-by-step guide into the root README.md.

3. Deleted the separate Markdown file from the chocolate folder.

TimeTriggerx committed:
Fix image format for drag and drop method

Updated image references for drag and drop method in README.

koppor reviewed (changes requested):
Content not helpful.

Provide link to the **correct** pdf belonging to the entry. Not an arbitrary rubbish PDF.

TimeTriggerx committed:
Fix: Replace placeholder PDFs with correct open-access papers for chocolate library

TimeTriggerx commented:
Hi @koppor, I sincerely apologize for the placeholder files. I mistakenly used 'dummy' PDFs because I was over-cautious about copyright issues and overlooked the fact that these specific entries are Open Access.

I have now corrected this by:

1. Locating the actual Open Access PDFs that correspond to the entries in chocolate.bib.

2. Replacing the placeholder files with these correct, authentic research papers.

3. Updating the .bib file to ensure the links are functional and the paths are relative.

I am here to learn and contribute meaningfully to JabRef. Thank you for your patience while I correct this.

TimeTriggerx committed:
docs: reorder open-access PDF entries alphabetically

koppor committed:
Discard changes to README.md

koppor committed:
Delete obsolete files

koppor committed:
Fix linebreak

koppor committed:
Rewrite using JabRef

koppor reviewed (approved):
You did even better: Attach multiple PDFs - with source. Thank you for the effort!

The .bib file was somehow changed with another application, because I needed to fix `groups-search-syntax` thingy.
